state/redis/redis.go

```diff
@@ -141,7 +141,7 @@ func (r *StateStore) Init(metadata state.Metadata) error {
 		r.client = r.newClient(m)
 	}
 
-	if _, err = r.client.Ping().Result(); err != nil {
+	if _, err = r.client.Ping(context.Background()).Result(); err != nil {
```

Comment:
I think it makes sense to add a `ctx` context variable in the `Redis` struct to handle cancelling the child contexts when closing the component. `r.ctx` would replace anywhere `context.Background()` is used. See the pubsub component.

Reply:
I've add ctx and cancel the same as the pubsub component.Also implement `Close() error` function.